### PR TITLE
Jc cf initial views

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delineated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* `react-router-dom` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

Notes:

* Some pairs have gone straight to wanting to integrate a CSS framework to achieve the “links pinned to the bottom of the page” AC here. You're encouraged to keep it simple at this point. Perhaps look into `position: fixed;` or perhaps look into a tutorial on CSS grid. There will be time to get fancy later in the project.

### PR DESCRIPTION
**Pull request by jmc617**
## Description
This PR adds React Router as well as the basic components of the app to the project.
### Initial Routes
We added [react-router-dom](https://www.npmjs.com/package/react-router-dom) as a dependency in the app. React Router is a library used to create routes in React. We created routes for the home page ("list" view) and /add-an-item ( "add an item" view). 
### Initial Components
We created the basic components to go along with the new routes. We used [Navlinks](https://reactrouter.com/docs/en/v6/api#navlink), from React Router, to create links and determine which link is active and make the link bold. We used inline CSS (`position:fixed` and `bottom:0`) to place the links at the bottom of the page.

## Related Issue

Closes #2 

## Acceptance Criteria

- [x] react-router-dom has been added as a project dependency
- [x] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- [x] When one of the links is clicked, the browser URL updates to represent the current view, and a matching view component is displayed
- [x] Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|  ✓ | :link: Update dependencies |

### Before
No routes or views.
### After
Two views with links that toggle between them. The links are at the bottom and the active link is bold.
Screenshot 2022-04-05 214833
Screenshot 2022-04-05 215021

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin jc-cf-initial-views` and check that branch out with `git checkout jc-cf-initial-views`
- Then use `npm install`  to install the newly added dependency locally and `npm start` to launch the app.
- Check that the links are present at the bottom of the page. Click the links and make sure the view text matches the URL; the home page should say "List of Items" and the /add-an-item page should say "Add an Item". Also, check that the link for the active page is bold.

### CODE REVIEW

**Review by HonduranCoder — COMMENTED** [review]
Good work. The one thing I noticed, we briefly discussed refactoring the code to the different file format. This would change the app.js stuff to be moved to two pages (a list item page & an add item page) inside the pages folder. Let me know if you have questions. Thanks!

**Comment by jmc617** [comment]
Yes @HonduranCoder, we believe that the simplest way to bring our code into the file structure is to merge both commits with main, and then @colefortner and I can separate our code into the structure. We are open to other ideas, we just didn't want to duplicate your system.

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,55 @@
...
+        to="/"
+        style={({ isActive }) =>
+          isActive ? { fontWeight: 'bold' } : { fontWeight: 'normal' }
+        }
+      >
+        List of Items
+      </NavLink>
+      <NavLink
+        to="/add-an-item"
+        style={({ isActive }) =>
```
Comment (MikeBLambert) [inline src/App.js:34]: Noice! I didn't know that `NavLink`'s `style` prop can take a callback like this! 

It looks like it [also takes an `activeStyle`](https://v5.reactrouter.com/web/api/NavLink/activestyle-object) too. So cool!

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,55 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>Welcome to React Router!</h1>
```
Comment (MikeBLambert) [inline src/App.js]: Maybe we want to say something more along the lines of "Welcome to your Smart Shopping List!"?

**Review by MikeBLambert — APPROVED** [review]
Really great work! Hits all of the AC. :-)

File: src/App.js (hunk 2 above)
Comment (jmc617) [reply src/App.js]: Thank you for the review! We changed the title.

**Pull request merged**